pkg/machines/components/vnc.jsx

```diff
@@ -83,16 +105,69 @@ class Vnc extends React.Component {
         console.error('VncConsole failed to init: ', detail, this);
     }
 
+    onExtraKeysDropdownToggle() {
+        this.setState({
+            isActionOpen: !this.state.isActionOpen
```

Comment:
This should use `this.setState((state, props) => ...`

Reply:
And this is called only when someone clicks on button from the dropdown, right? Then it can just close it.